common/http/src/metrics_endpoint.rs

```diff
@@ -0,0 +1,39 @@
+use futures::TryFutureExt;
+use hyper::{
+    header::CONTENT_TYPE,
+    service::{make_service_fn, service_fn},
+    Body, Request, Response, Server,
+};
+use prometheus::{Encoder, TextEncoder};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error(transparent)]
+    Server(#[from] hyper::Error),
+}
+
+pub async fn serve(port: &u32) -> Result<(), Error> {
+    let address = format!("0.0.0.0:{}", port).parse().unwrap();
```

Comment:
Again extremely minor nit

going through the format and socket addr parse machinery then unwrapping to get a socketaddr feels awkward when we could do something like
`
SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), port)
`
even if the unwrap can't fail